docs/cookbook/languages/python.md

```diff
@@ -0,0 +1,289 @@
+
+To find these Python packages in the Flox Catalog, use the same `flox search` syntax from before:
+
+```
+flox search numpy
+```
+
+This will return the set of packages that match your search term:
+
+```
+$ flox search numpy
+[...]
+python311Packages.numpy    Scientific tools for Python
+[...]
+```
+
+Python packages will have a prefix of `pythonXXXPackages.` in their package name. To install them, use the entire package name including the prefix:
+
+```
+flox install python311Packages.numpy python311Packages.pandas
+```
+
+You will see output indicating that the packages were successfully installed. If installation was not successful, you should see an error message indicating the failure.
+
+```
+$ flox install python311Packages.numpy python311Packages.pandas
+✅ 'numpy' installed to environment 'new-python-project'
+✅ 'pandas' installed to environment 'new-python-project'
+```
+
+### Activate and verify packages
+
+Once the packages have been installed, activate the new environment:
+
+```
+flox activate
+```
+
+This will put you into a new subshell with your environment active:
+
+```
+$ flox activate
+✅ You are now using the environment 'new-python-project'.
+To stop using this environment, type 'exit'
+
+flox [new-python-project] $
+```
+
+## Using Flox in an existing project
+
+If you are working with an existing project that is already configured to work with Python - e.g. it has a `requirements.txt` or `pyproject.toml` - Flox provides an automated environment setup flow.
+
+For this example we will clone the `eralchemy` repo, which already contains Python configuration:
+
+```
+git clone https://github.com/eralchemy/eralchemy.git && cd eralchemy
+```
+
+### Auto-initialize the environment
+
+Once inside `eralchemy`s project directory, initialize a new Flox environment:
+
+```
+flox init
+```
+
+You will see the following question in the output:
+
+```
+$ flox init
+Flox detected a Python project with the following Python provider(s):
+
+* pyproject (generic pyproject.toml)
+
+  Installs python (3.12.5) with pip bundled.
+  Adds a hook to setup a venv.
+  Installs the dependencies from the pyproject.toml to the venv.
+
+! Would you like Flox to set up a standard Python environment?
+You can always change the environment's manifest with 'flox edit'
+  Yes - with pyproject
+  No
+> Show suggested modifications
+[Use '--auto-setup' to apply Flox recommendations in the future.]
+```
+
+When you initialize Flox and it finds a `pyproject.toml` or `requirements.txt` file inside a project directory, it automatically runs this wizard.
+
+Here's what each option does:
+
+- “Yes” builds the environment using the `pyproject.toml` file;
+- “No” skips automatic setup. You can use `pip` or `poetry` with `pyproject.toml` to build your environment;
+- “Show” previews the configuration you'd get by selecting “Yes,” letting you vet your environment's setup.
+
+And here's what happens if you choose the “Show” option:
```

Comment:
```suggestion

If you would like to preview the proposed changes, you can choose the "Show" option. It will show you the proposed changes to each section of your environment manifest, e.g.:
```